app/main.py

```diff
@@ -0,0 +1,202 @@
+"""
+Module which runs the GUI for the project using FastAPI
+Copyright (C) 2020  Royston E Tauro & Sammith S Bharadwaj & Shreyas Raviprasad
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as published
+    by the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+"""
+
+import time
+import os
+from fastapi.templating import Jinja2Templates
+from fastapi.responses import HTMLResponse, RedirectResponse, FileResponse
+from fastapi import FastAPI, Form, Request, status, BackgroundTasks
+from fastapi.staticfiles import StaticFiles
+from fastapi.encoders import jsonable_encoder
+from app.fantasy_cricket.team import Teams
+from app.fantasy_cricket.utils import Matches
+from crawler.spiders.howstat import HowstatSpider
+
+# pylint: disable=missing-function-docstring
+# pylint: disable=global-variable-undefined
+
+app = FastAPI()
+
+templates = Jinja2Templates(directory="./app/fantasy_cricket/templates")
+app.mount("/static", StaticFiles(directory="./app/fantasy_cricket/static"), name="static")
+
+
+cricket = Matches()
+
+
+@app.get("/", response_class=HTMLResponse)
+def home(request: Request):
+    matches = cricket.get_match()
+    teams = [
+        [match[0]["name"], match[1]["name"], match[2]["flag"], match[3]["flag"]]
+        for match in matches
+    ]
+    return templates.TemplateResponse(
+        "index.html", {"request": request, "teams": teams}
+    )
+
+
+@app.post("/")
+async def home_post(match: str = Form(...)):
+    response = RedirectResponse(
+        url="/playing11?team1="
+        + match.split(" vs ")[0]
+        + "&team2="
+        + match.split(" vs ")[-1],
+        status_code=status.HTTP_302_FOUND,
+    )
+    return response
+
+
+@app.get("/playing11", response_class=HTMLResponse)
+def playing_11(request: Request,team1,team2):
```

Comment:
Any reason why some endpoints are async and some aren't?

Reply:
This need not be async
But the playing11_post will need to be async cause of the await on request.form